(GH-1888) Create separate reference docs for config files (#1923)
puppetlabs/bolt: documentation/bolt_inventory_reference.md

# `inventory.yaml` options

The [inventory file](inventory_file_v2.md) is used to store information about
your targets and arrange them into groups. Both targets and groups can
include data such as facts or vars and can configure the transports Bolt uses
to establish connections with targets.

> ## Options

Reviewer: ```suggestion
## Top-level options

The top level of your inventory file contains options that configure the implicit `all` group. These options apply to all of the groups and targets and targets in the inventory file. For more information on inventory files, such as precedence, see [Inventory files](inventory_file_v2.md)
```
The next heading is "Group objects", but obviously contains some identical options. We need something to make it clearer that all of these options are for the `all` group. I think top-level is fine, but I'm open to suggestions. I think adding the short intro paragraph is also helpful

Reviewer 2: These are more like _fields_ than _options_ because this file is more akin to data than configuration.

- **Type:** Hash

```yaml
facts:
  os:
    family: Darwin
```

### `features`

A list of available features for the implicit `all` group. Features set at this
level apply to all groups and targets in the inventory file.

- **Type:** Array

> 🔩 **Tip:** You can set the `puppet_agent` feature to indicate that Bolt
> should skip installing the Puppet agent on all targets when using
> `bolt apply` or the `apply_prep` plan function.

```yaml
features:
  - puppet_agent
```

### `groups`

A [list of groups](#group-objects) and their associated configuration.

- **Type:** Array

```yaml
groups:
  - name: linux
    targets:
      - linux-1.example.com
      - linux-2.example.com
  - name: windows
    targets:
      - windows-1.example.com
      - windows-2.example.com
```

### `targets`

A [list of targets](#target-objects) and their associated configuration.

- **Type:** Array

```yaml
targets:
  - target1.example.com
  - target2.example.com
```

### `vars`

A map of variables for the implicit `all` group. Variables set at this level
apply to all groups and targets in the inventory file.

- **Type:** Hash

```yaml
vars:
  ssh_config: /etc/ssh_config
```

## Group objects

> The `groups` key is used to specify a list of groups, which contain a list of

Reviewer: ```suggestion
Use a `groups` key to specify a list of groups, which contain a list of
```

targets. Each item in the `groups` list is a map of data and configuration for
the group. Group objects accept many of the same keys as the implicit `all`
group.

### `config`

A map of configuration options for the group. Configuration set at this level

> applies to all groups and targets under the group. A detailed description of
> each option, their default values, and any available sub-options can be viewed
> in [Transport configuration reference](#bolt_transports_reference.md).

Reviewer: ```suggestion
applies to all groups and targets under the group. For a detailed description of
each option, their default values, and any available sub-options,
see [Transport configuration reference](#bolt_transports_reference.md).
```

- **Type:** Array

```yaml
groups:
  - name: linux
    targets:
      - linux-1.example.com
      - linux-2.example.com
```

### `vars`

A map of variables for the group. Vars set at this level apply to all groups
and targets under the group.

- **Type:** Hash

```yaml
groups:
  - name: linux
    vars:
      ssh_config: /etc/ssh_config
```

## Target objects

The `targets` key is used to specify a list of targets. Each item in the
`targets` list must be one of the following:

- A string representation of the target's URI
- A map of data and configuration options for the target

When specifying a target using a map of data and configuration, the following
options are available:

### `alias`

A unique alias to refer to the target. Aliases cannot conflict with the name
of a group, the name of a target, or another alias.

- **Type** Array

```yaml
targets:
  - uri: linux-1.example.com
    alias:
      - database
  - uri: linux-2.example.com
    alias:
      - webserver
```

### `config`

A map of configuration options for the target. A detailed description of each
option, their default values, and any available sub-options can be viewed in

> [Transport configuration reference](#bolt_transports_reference.md).

Reviewer: ```suggestion
[Transport configuration reference](bolt_transports_reference.md).
```

    config:
      transport: winrm
      winrm:
        user: bolt
        password: hunter2!
```

### `facts`

A map of system information, also known as
[facts](https://puppet.com/docs/puppet/latest/lang_facts_and_builtin_vars.html),
for the target.

- **Type:** Hash

```yaml
targets:
  - uri: linux.example.com
    facts:
      os:
        architecture: x86_64
```

### `features`

A list of available features for the target.

- **Type:** Array

> 🔩 **Tip:** You can set the `puppet_agent` feature on the target to indicate
> that Bolt should skip installing the Puppet agent on the target when using
> `bolt apply` or the `apply_prep` plan function.

```yaml
targets:
  - uri: linux.example.com
    features:
      - puppet_agent
```

### `name`

A human-readable name for the target. This option is required unless the `uri`
option is set.

- **Type:** String

```yaml
targets:
  - name: database
```

### `uri`

The URI of the target. This option is required unless the `name` option is set.

- **Type:** String

```yaml
targets:
  - uri: linux.example.com
```

### `vars`

A map of variables for the target.

- **Type:** Hash

```yaml
targets:
  - uri: linux.example.com
    vars:

>       ssh_config: /etc/ssh_config

Reviewer 2: Missing ``` here causes the example file to be part of this code block and to not appear in the table of contents.

Author: Updated.
